src/bitcoin-cli.cpp

```diff
@@ -291,6 +295,211 @@ class GetinfoRequestHandler: public BaseRequestHandler
     }
 };
 
+/** Process netinfo requests */
+class NetinfoRequestHandler : public BaseRequestHandler
+{
+private:
+    bool IsAddrIPv6(const std::string& addr) const
+    {
+        return !addr.empty() && addr.front() == '[';
+    }
+    bool IsInboundOnion(const std::string& addr_local, int mapped_as) const
+    {
+        return mapped_as == 0 && addr_local.find(ONION) != std::string::npos;
+    }
+    bool IsOutboundOnion(const std::string& addr, int mapped_as) const
+    {
+        const size_t addr_len{addr.size()};
+        const size_t onion_pos{addr.rfind(ONION)};
+        return mapped_as == 0 && onion_pos != std::string::npos && addr_len > ONION_LEN &&
+               (onion_pos == addr_len - ONION_LEN || onion_pos == addr.find_last_of(":") - ONION_LEN);
+    }
+    uint8_t m_details_level{0}; //!< Optional user-supplied arg to set dashboard details level
+    bool DetailsRequested() const { return m_details_level > 0 && m_details_level < 5; }
+    bool IsAddressSelected() const { return m_details_level == 2 || m_details_level == 4; }
```

Comment:
Going from details level 2 to 3, it is indeed increasingly detailed as the address is replaced by 2 fields combined in one, version and sub-version. That said, I don't mind improving it if there is a good suggestion.

Reply:
(As there may be people who prefer to see one or the other of these two wider fields, but not necessarily both for reasons of window size or interest, and since it isn't complicated to do--and is done--it's good to allow people the choice.)

Reply:
FWIW I also found it confusing that increasing what I thought was a numeric verbosity level removed columns :)

I think the expected command-line UI for verbosity switches are one of these two:
* Increase verbosity like for `curl`: `curl -v`, `curl -vv`, `curl -vvv`, etc.
* Specify exactly what you want like for `ps`: `ps -o comm,label`

Reply:
Concrete suggestion for the help?

Reply:
How about "An optional integer argument from 0 to 4 can be passed for different peers listings (default: 0)"

Reply:
Yes, that is probably good enough for now. Making the verbosity levels more in tune with how other command-line tools work can be done in a follow-up PR: no need to do it here :)

Reply:
Passing args like for `curl` or `ps` would be indeed cool, but yes, overkill for now and maybe hurt this being merged by making it too large.

Reply:
Done. Thanks for the feedback. The helps seems better now.